### PR TITLE
Fix storing of orderlistidcounter in Json

### PR DESCRIPTION
**Pull request by nigel27022001**

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/CompleteOrderCommand.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+        CompleteOrderCommand otherCompleteOrderCommand = (CompleteOrderCommand) other;
+        return targetIndex.equals(otherCompleteOrderCommand.targetIndex);
+    }
+
+    @Override
+    public String toString() {
+        return new ToStringBuilder(this)
+                .add("targetIndex", targetIndex)
+                .toString();
```
Comment (chonghaoen) [inline src/main/java/seedu/address/logic/commands/CompleteOrderCommand.java:62]: would one line be cleaner?

File: src/main/java/seedu/address/logic/commands/CompleteOrderCommand.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.order.Order;
+
+/**
+ * Deletes an order by its Order ID.
```
Comment (chonghaoen) [inline src/main/java/seedu/address/logic/commands/CompleteOrderCommand.java:13]: change?

**Review by chonghaoen — COMMENTED** [review]
some things to change

File: src/main/java/seedu/address/logic/commands/CompleteOrderCommand.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    public static final String MESSAGE_COMPLETE_ORDER_SUCCESS = "Completed Order: %1$s";
+
+    private final Index targetIndex;
+
+    public CompleteOrderCommand(Index targetIndex) {
+        this.targetIndex = targetIndex;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
```
Comment (bertrandong) [inline src/main/java/seedu/address/logic/commands/CompleteOrderCommand.java:32]: Good implementation!

**Review by bertrandong — COMMENTED** [review]
LGTM, remember to test the code you have written.

File: src/main/java/seedu/address/logic/commands/CompleteOrderCommand.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        requireNonNull(model);
+        String successStr = "";
+        for (Index targetIndex : targetIndexArr) {
+            if (!model.orderIdExists(targetIndex.getOneBased())) {
+                throw new CommandException(Messages.MESSAGE_INVALID_ORDER_DISPLAYED_INDEX);
+            }
+            Order completedOrder = model.findOrderByIndex(targetIndex.getOneBased());
+            model.deleteOrder(targetIndex.getOneBased());
+            successStr = successStr + ", " + targetIndex.getOneBased();
+        }
```
Comment (Joelwang22) [inline src/main/java/seedu/address/logic/commands/CompleteOrderCommand.java:44]: Great Implementation!

File: src/main/java/seedu/address/logic/parser/CompleteOrderCommandParser.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+            String[] nameKeywords = trimmedArgs.split("\\s+");
+            ArrayList<Index> indexArr = new ArrayList<>();
+            for (String s : nameKeywords) {
+                Index index = ParserUtil.parseIndex(s);
+                indexArr.add(index);
+            }
+            return new CompleteOrderCommand(indexArr);
+        } catch (ParseException pe) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, CompleteOrderCommand.MESSAGE_USAGE), pe);
```
Comment (Joelwang22) [inline src/main/java/seedu/address/logic/parser/CompleteOrderCommandParser.java:31]: Great Implementation!

File: src/main/java/seedu/address/model/AddressBook.java [hunk 6]
```diff
@@ -190,9 +207,24 @@ public void removeOrder(int id) {
...
 
+    public void completeOrder(int id) {
+        Order orderToComplete = this.findOrderByIndex(id);
+        Person editedCustomer = orderToComplete.getCustomer();
+        ArrayList<Order> listToEdit = editedCustomer.getOrders();
+        listToEdit.remove(orderToComplete);
+        editedCustomer.setOrders(listToEdit);
+        this.setPerson(orderToComplete.getCustomer(), editedCustomer);
+
+        activeOrders.deleteOrder(id);
```
Comment (Joelwang22) [inline src/main/java/seedu/address/model/AddressBook.java:221]: Great Implementation!

File: src/main/java/seedu/address/storage/CSVCompletedOrderStorage.java [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+        return filePath;
+    }
+
+    public void saveCompletedOrder(ReadOnlyAddressBook addressBook, Path filePath) throws IOException {
+        requireNonNull(addressBook);
+        requireNonNull(filePath);
+
+        FileUtil.createIfMissing(filePath);
+        ObservableList<Order> completedOrderList = addressBook.getCompletedOrderList();
+        List<String[]> dataToBeStored = convertOrdersToReadable(completedOrderList);
```
Comment (Joelwang22) [inline src/main/java/seedu/address/storage/CSVCompletedOrderStorage.java:44]: Great Implementation!

File: src/main/java/seedu/address/storage/CSVCompletedOrderStorage.java [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+    public String[] convertProductMapToReadable(Order order, Product product) {
+        requireNonNull(order);
+        requireNonNull(product);
+
+        ArrayList<String> orderDetails = new ArrayList<>();
+        orderDetails.add(convertOrderIdToReadable(order));
+        orderDetails.add(convertOrderCustomerToReadable(order));
+        orderDetails.add(convertProductToReadable(product));
+        orderDetails.add(convertQuantityToReadable(order, product));
+        return orderDetails.toArray(STRING_TYPE_SPECIFIER);
```
Comment (Joelwang22) [inline src/main/java/seedu/address/storage/CSVCompletedOrderStorage.java:65]: Great Implementation!

**Pull request merged**